gquant/dataframe_flow/task.py

```diff
@@ -126,69 +178,52 @@ def get_node_obj(self, replace=None, profile=False, tgraph_mixin=False):
+                module_dir = loaded.path
+                mod = loaded.mod
                 NodeClass = getattr(mod, node_type)
             else:
-                NodeClass = None
                 try:
                     global DEFAULT_MODULE
                     plugmod = os.getenv('GQUANT_PLUGIN_MODULE', DEFAULT_MODULE)
                     MODLIB = importlib.import_module(plugmod)
                     NodeClass = getattr(MODLIB, node_type)
                 except AttributeError:
                     modules = get_gquant_config_modules()
-                    if (module_name is not None):
-                        loaded = load_modules(
-                            modules[module_name], name=module_name)
+                    for key in modules:
+                        loaded = load_modules(modules[key], name=key)
                         module_dir = loaded.path
                         mod = loaded.mod
                         try:
                             NodeClass = getattr(mod, node_type)
+                            break
                         except AttributeError:
-                            pass
-                    else:
-                        for key in modules:
-                            loaded = load_modules(modules[key], name=key)
-                            module_dir = loaded.path
-                            mod = loaded.mod
-                            try:
-                                NodeClass = getattr(mod, node_type)
-                                break
-                            except AttributeError:
-                                continue
-                    if NodeClass is None:
-                        raise Exception("Cannot find the Node Class:" +
-                                        node_type)
-                    else:
-                        append_path(module_dir)
-                        try:
-                            # add python path to all the client workers
-                            # assume all the worikers share the same directory
-                            # structure
-                            import dask.distributed
-                            client = dask.distributed.client.default_client()
-                            client.run(append_path, module_dir)
-                        except (ValueError, ImportError):
-                            pass
+                            continue
+            if NodeClass is None:
+                raise Exception("Cannot find the Node Class:" +
+                                node_type)
+            if module_dir:
+                append_path(module_dir)
+                try:
+                    # add python path to all the client workers
+                    # assume all the worikers share the same directory
+                    # structure
+                    import dask.distributed
+                    client = dask.distributed.client.default_client()
+                    client.run(append_path, module_dir)
```

Comment:
This part is doing something similar to 
```python

        def append_path(worker):
            for ipath in all_paths:
                if ipath not in sys.path:
                    sys.path.append(ipath)

        # TODO: This could be a Ray Driver functionality. Add module paths
        #     that might have been imported in
        #         task.py:Task:get_node_obj
        ray.worker.global_worker.run_function_on_all_workers(append_path)
```
Does it make sense to consolidate them in one place?

Reply:
Yes, it makes sense. Really, I think this type of configuration logic belongs in a "driver" context i.e. our issue (https://github.com/rapidsai/gQuant/issues/96). Ray has this concept of a driver. The Dask logic should also go into a "dask driver" node. In my opinion having this dask code in task.py is temporary until we flush out this "driver" idea.

I actually placed the ray code path append code initially there under "task.py" as well. But doing that for whatever reason caused ray to do everything on 1 GPU instead of using all the GPUs on the system. Ray is still a mystery to me. I understand Dask a lot better than Ray.

Reply:
I tested the code to append path to sys for ray in task.py similar to how its done for dask:
```
            if module_dir:
                append_path(module_dir)
                try:
                    # add python path to all the client workers
                    # assume all the worikers share the same directory
                    # structure
                    import dask.distributed
                    client = dask.distributed.client.default_client()
                    client.run(append_path, module_dir)
                except (ValueError, ImportError):
                    pass

                try:
                    import ray

                    def ray_append_path(worker):
                        import sys
                        if module_dir not in sys.path:
                            sys.path.append(module_dir)

                    # TODO: This could be a Ray Driver functionality. Add
                    #     module paths that might have been imported in
                    ray.worker.global_worker.run_function_on_all_workers(
                        ray_append_path)
                except (ValueError, ImportError):
                    pass
```

This worked perfectly fine. I'm not sure why it didn't work before. I can change the code and remove this logic from `NemoHyperTuneNode`.